src/util/futils.h

```diff
@@ -25,7 +25,7 @@ extern int git_futils_readbuffer(git_str *obj, const char *path);
 extern int git_futils_readbuffer_updated(
 	git_str *obj,
 	const char *path,
-	unsigned char checksum[GIT_HASH_SHA1_SIZE],
+	unsigned char checksum[GIT_HASH_SHA256_SIZE],
```

Comment:
```suggestion
	unsigned char checksum[GIT_HASH_MAX_SIZE],
```

Reply:
Currently `git_futils_readbuffer_updated()` is very strict in its usage of SHA256.
It calculates a new hash using `GIT_HASH_ALGORITHM_SHA256` and updates the passed checksum buffer.
Changing it to `GIT_HASH_MAX_SIZE` should be "safe" since its always larger, but calling it with any checksum other than SHA256 will fail.
Maybe we should pass in a `git_hash_ctx *` instead and make `git_futils_readbuffer_updated()` sensitive to the type of hash used?

From what I can tell it is only used by `git_grafts_refresh()` that uses SHA256 to track changes in a private struct unrelated to what the repository uses.

Reply:
Oops, yes, you're absolutely right. Thanks for the clarification.